darts/models/theta.py

```diff
@@ -115,14 +120,241 @@ def predict(self, n: int) -> 'TimeSeries':
+                self.model_mode = Model.ADDITIVE
+            theta_t = self.theta * ts_values + (1 - self.theta) * theta0_in
+
+        # SES part of the decomposition.
+        self.model = hw.SimpleExpSmoothing(theta_t).fit()
+        theta2_in = self.model.fittedvalues
+
+        if (theta2_in > 0).all() and self.model_mode is Model.MULTIPLICATIVE:
+            self.fitted_values = theta2_in**self.wses * theta0_in**self.wdrift
+        else:
+            if self.model_mode is Model.MULTIPLICATIVE:
+                self.model_mode = Model.ADDITIVE
+                logger.warn("Negative Theta line. Fallback to additive model")
+                theta_t = self.theta * new_ts.values() + (1 - self.theta) * theta0_in
+                self.model = hw.SimpleExpSmoothing(theta_t).fit()
+                theta2_in = self.model.fittedvalues
+            self.fitted_values = self.wses * theta2_in + self.wdrift * theta0_in
+        if self.is_seasonal:
+            if self.season_mode is Season.ADDITIVE:
+                self.fitted_values += self.seasonality.univariate_values()
+            elif self.season_mode is Season.MULTIPLICATIVE:
+                self.fitted_values *= self.seasonality.univariate_values()
+        # Fitted values are the results of the fit of the model on the train series. A good fit of the model
+        # will lead to fitted_values similar to ts. But one cannot see if it overfits.
+        if self.normalization:
+            self.fitted_values *= self.mean
+        # Takes too much time to create a TimeSeries
+        # Overhead: 30% ± 10 (2-10 ms in average)
+        self.fitted_values = TimeSeries.from_times_and_values(ts.time_index(), self.fitted_values)
+
+    def predict(self, n: int) -> 'TimeSeries':
+        super().predict(n)
+
+        # Forecast of the SES part.
+        forecast = self.model.forecast(n)
+
+        # Forecast of the Linear Regression part.
+        drift = self.drift(np.arange(self.length, self.length + n))
+        if self.trend_mode is Trend.EXPONENTIAL:
+            drift = np.exp(drift)
+
+        if self.model_mode is Model.ADDITIVE:
+            forecast = self.wses * forecast + self.wdrift * drift
+        else:
+            forecast = forecast**self.wses * drift**self.wdrift
+
+        # Re-apply the seasonal trend of the TimeSeries
+        if self.is_seasonal:
+
+            replicated_seasonality = np.tile(self.seasonality.pd_series()[-self.season_period:],
+                                             math.ceil(n / self.season_period))[:n]
+            if self.season_mode is Season.MULTIPLICATIVE:
+                forecast *= replicated_seasonality
+            else:
+                forecast += replicated_seasonality
+
+        if self.normalization:
+            forecast *= self.mean
+
+        return self._build_forecast_series(forecast)
+
+    @staticmethod
+    def select_best_model(ts: TimeSeries, thetas: Optional[List[int]] = None,
```

Comment:
We can keep it temporarily, but then eventually I think we should have each model (if possible) given good reasonable default hyper parameters sets, and then let the user call `backtest_gridsearch(params='default', ...)` or something like this.
Ping @grll